### ISSUE TITLE
Import are removed although they are used

### ISSUE THREAD

**Issue by buehler** [cybertim/vscode-gengetset#24]
Hey @cybertim 

I'm really happy with your extension, it works (more or less) like it shoud :wink:

Although, I have one small issue which makes it unusable :-(

I use [giuseppe](https://github.com/smartive/giuseppe) for my api which makes extensive use of typescript decorators. Now I've got some decorators that use imports, and they are removed from the imports list.
example:

``` typescript
import {/*...*/, isNumber} from 'giuseppe';

@Controller()
class Ctrl {
    @Get()
    public foobar(@Query('foobar', {validator: isNumber()}) queryParam: any): any{
        /*...*/
    }
} 
```

In the example above, `isNumber` is clearly use, but when I add another import or use "optimize imports" the reference to `isNumber` is removed.

Cheers and thanks for your work!
Chris

**Comment by buehler** [comment]
Actually, I found the code that is preventing the import from working:
https://github.com/cybertim/vscode-gengetset/blob/master/src/import.ts#L317

This line does ignore the word. I assume there is some serious thinking behind this logic, but why do you ignore such functions? There are possibly many libraries that are using functions and types like that :-)

Or do I miss something?

Cheers

Edit:
my suggestion would be to check if the function is explicitly exported and accept it then.

**Comment by cybertim** [comment]
Nice to hear you're happy with the extension @buehler :)

Regarding the `commonKeywordStartsWith` this is (as is a lot of code in this extension) a workaround to keep a lot of internal exports for some complex libraries (like Angular) out of the scope.

Since there is no real definition yet of how a library created for NPM with typescript and how it should be analysed I went with the most recent and common approach eg. indexing everything and search for the `index.ts` later to create a nice import line. So naturally this will also extract a lot of internal components.

A better way would be searching for the `index.ts` first and then analysing all imported lines. But this will introduce another level of recursion and complexity and code .. so when the dust settles and there are real rules about how libraries should be created with typescript I will probably add this.

So I checked some popular libraries (Angular and Ionic) and they used a lot of lowercased words to define their internal components hence the weird `commonKeywordStartsWith` list.

And thanks for the pr, I will check it when I have time to test it against a few libraries.

Also I could just push the list to the settings so it is editable and add a boolean setting to disable it completely until a better alternative indexing is available.
